tiling_segmentation.py

```diff
@@ -184,55 +186,169 @@ def make_dataset_file_name(exp_name: str, min_annot: Number, dataset: str, attr_
         dataset_file_name = f'{exp_name}{sampling_str}_{dataset}.csv'
         return dataset_file_name, sampling_str
 
+    @staticmethod
+    def _save_tile(sample, dst, window, crs):
+        """
+        Save individual raster patch as a geotiff.
+        Args:
+            sample: numpy array with a shape of (h, w, c).
+            dst: destination file path.
+            window: bounding box coordinates of the patch.
+            crs: patch's crs.
+
+        Returns: None
+        """
+        # Unpack the bounding box coordinates and the sample's dimensions.
+        xmin, ymax, xmax, ymin = window
+        n_rows, n_cols, n_bands = sample.shape[1], sample.shape[2], sample.shape[0]
+
+        # Calculate spatial resolution based on the extracted data:
+        xres = (xmax - xmin) / float(n_cols)
+        yres = (ymax - ymin) / float(n_rows)
+        # Set the geotransformation parameters in the GDAL format:
+        geotransform = (xmin, xres, 0, ymax, 0, -yres)
+
+        # Create an output raster patch datasource:
+        drv = gdal.GetDriverByName("GTiff")
+        gdal_type = gdal.GDT_Byte if sample.dtype == np.uint8 else gdal.GDT_UInt16
+        dst_ds = drv.Create(dst, n_cols, n_rows, n_bands, gdal_type, options=["COMPRESS=DEFLATE"])
```

Comment:
I'd favor a rasterio based implementation here. Can stay as is for this PR though. I'm ok with this as long as we don't get problems using rasterio and gdal in the same script: https://rasterio.readthedocs.io/en/latest/topics/switch.html

Reply:
Raster based causes PermissionError in some cases so I used gdal here.